### PR TITLE
fix(detected_object_validation): use search for searching intersected lanelet

### PR DESCRIPTION
**Pull request by a-maumau**
## Description
This PR is to use the [`search`](https://github.com/fzi-forschungszentrum-informatik/Lanelet2/blob/4789fde6c3bafcd5a8d721d11c5d92870ca59a22/lanelet2_core/include/lanelet2_core/LaneletMap.h#L172) method instead of searching the intersected lanelets by iterating with a for loop.

Changes:
- `getIntersectedLanelets`: change to use `search`
- `objectCallback`: merge to single call of `getIntersectedLanelets`,
- `filterObject`: merge conditions of `intersected_road_lanelets` and `intersected_shoulder_lanelets`
- `lanelet_filter.hpp`: remove unused var and fix method args

## Related links

**Parent Issue:**

- Link

## How was this PR tested?
Checked by comparing the fetched intersected lanelet number in the autoware [sample scene](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/).
- not checked the lanelets are completely same.

Check processing time of `getIntersectedLanelets`:
- not perfectly aligned, near the same timestamp
- marginal improvement for a small scene (expected)

#### original
```Text
...
[object_lanelet_filter_node-53] process_time: 116 microseconds
[object_lanelet_filter_node-53] process_time: 113 microseconds
[object_lanelet_filter_node-53] process_time: 108 microseconds
[object_lanelet_filter_node-53] process_time: 199 microseconds
[object_lanelet_filter_node-53] process_time: 113 microseconds
[object_lanelet_filter_node-53] process_time: 109 microseconds
[object_lanelet_filter_node-53] process_time: 108 microseconds
[object_lanelet_filter_node-53] process_time: 111 microseconds
[object_lanelet_filter_node-53] process_time: 127 microseconds
[object_lanelet_filter_node-53] process_time: 114 microseconds
[object_lanelet_filter_node-53] process_time: 113 microseconds
[object_lanelet_filter_node-53] process_time: 108 microseconds
[object_lanelet_filter_node-53] process_time: 111 microseconds
[object_lanelet_filter_node-53] process_time: 214 microseconds
[object_lanelet_filter_node-53] process_time: 119 microseconds
[object_lanelet_filter_node-53] process_time: 114 microseconds
...
```

#### after this PR
```Text
...
[object_lanelet_filter_node-53] process_time: 108 microseconds
[object_lanelet_filter_node-53] process_time: 106 microseconds
[object_lanelet_filter_node-53] process_time: 107 microseconds
[object_lanelet_filter_node-53] process_time: 120 microseconds
[object_lanelet_filter_node-53] process_time: 111 microseconds
[object_lanelet_filter_node-53] process_time: 108 microseconds
[object_lanelet_filter_node-53] process_time: 105 microseconds
[object_lanelet_filter_node-53] process_time: 110 microseconds
[object_lanelet_filter_node-53] process_time: 182 microseconds
[object_lanelet_filter_node-53] process_time: 139 microseconds
[object_lanelet_filter_node-53] process_time: 106 microseconds
[object_lanelet_filter_node-53] process_time: 101 microseconds
[object_lanelet_filter_node-53] process_time: 108 microseconds
[object_lanelet_filter_node-53] process_time: 116 microseconds
[object_lanelet_filter_node-53] process_time: 108 microseconds
[object_lanelet_filter_node-53] process_time: 115 microseconds
[object_lanelet_filter_node-53] process_time: 156 microseconds
...
```

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/detected_object_validation/src/lanelet_filter/lanelet_filter.cpp [hunk 1]
```diff
@@ -214,18 +209,41 @@ LinearRing2d ObjectLaneletFilterNode::getConvexHull(
...
+
+  lanelet::Lanelets candidates_lanelets = lanelet_map_ptr_->laneletLayer.search(bbox2d);
+  for (const auto & lanelet : candidates_lanelets) {
+    // only check the road lanelets and road shoulder lanelets
+    if (lanelet.hasAttribute(lanelet::AttributeName::Subtype) && (
+        lanelet.attribute(lanelet::AttributeName::Subtype).value() == lanelet::AttributeValueString::Road ||
+        lanelet.attribute(lanelet::AttributeName::Subtype).value() == "road_shoulder"
+      )
+    ) {
+      if (boost::geometry::intersects(convex_hull, lanelet.polygon2d().basicPolygon())) {
```
Comment (a-maumau) [inline perception/detected_object_validation/src/lanelet_filter/lanelet_filter.cpp:237]: not sure we need `boost::geometry::intersects` here.

**Comment by YoshiRi** [comment]
@a-maumau  cc: @technolojin 
Thanks for you PR. Could you test with larger lanelet_map such as odaiba_large_scale or something?

**Comment by technolojin** [comment]
test [TIER IV INTERNAL ](https://evaluation.tier4.jp/evaluation/reports/6bc14cbc-3d2c-53b1-96d6-dec43f26212a?project_id=prd_jt)

**Comment by technolojin** [comment]
@YoshiRi 
I tested under odaiba_large_scale with the issued data.
Overall CPU usage (under my environment) was dropped 10~20%

**Review by technolojin — APPROVED** [review]
LGTM

**Review by YoshiRi — APPROVED** [review]
LGTM

**Pull request merged**